InstaBotPrototype/DBMigrator/Program.cs

```diff
@@ -1,93 +1,42 @@
-                    }
-                    else
-                    {
-                        Console.WriteLine("Something went wrong...");
-                    }
-                }
-
-                Console.WriteLine(string.Empty.PadRight(30, '-'));
-            }
-        }
-
-        private static DBMigration[] GetMigrations()
-        {
-            List<DBMigration> list = new List<DBMigration>();
-
             string connString = ConfigurationManager.ConnectionStrings[1].ConnectionString;
             string provider = ConfigurationManager.ConnectionStrings[1].ProviderName;
 
-            DbProviderFactory factory = DbProviderFactories.GetFactory(provider);
+            var factory = DbProviderFactories.GetFactory(provider);
 
-            DbConnection dbConnection = factory.CreateConnection();
+            var dbConnection = factory.CreateConnection();
             dbConnection.ConnectionString = connString;
+            
 
-            DirectoryInfo directory = new DirectoryInfo(Directory.GetCurrentDirectory() + "/Migrations");
-
-            foreach (FileInfo file in directory.GetFiles("*.txt"))
-            {
-                using (StreamReader streamReader = new StreamReader(file.OpenRead()))
-                {
-                    DbCommand apply = factory.CreateCommand();
-                    apply.CommandText = streamReader.ReadLine();
-
-                    DbCommand reverse = factory.CreateCommand();
-                    reverse.CommandText = streamReader.ReadLine();
-
-                    string name = file.Name.Substring(0, file.Name.Length - 4);
+            var migrations = (from type in Assembly.GetCallingAssembly().GetTypes() where type.IsSubclassOf(typeof(DBMigration)) orderby type.GetCustomAttribute<IndexerAttribute>().Id select Activator.CreateInstance(type, new object[] { factory, dbConnection }) as DBMigration).ToList();
+            
 
-                    DBMigration migration = new DBMigration(apply, reverse, name) { Factory = factory, ConnectionString = connString };
-
-                    list.Add(migration);
-                }
-            }
-
-            dbConnection.Close();
-
-            DbCommand command = factory.CreateCommand();
+            var command = factory.CreateCommand();
             command.CommandText = "select * from dbo.Migrations";
-            command.Connection = factory.CreateConnection();
-            command.Connection.ConnectionString = connString;
+            command.Connection = dbConnection;
 
-            command.Connection.Open();
+            dbConnection.Open();
 
             try
             {
-                DbDataReader reader = command.ExecuteReader();
+                var reader = command.ExecuteReader();
 
                 while (reader.Read())
                 {
                     string name = (string)reader.GetValue(0);
 
-                    foreach (DBMigration migration in list)
+                    foreach (var migration in migrations)
```

Comment:
i'd recomend to extract check for migration to be applied into base class. its not good to fill IsApplied member in external function, better to incapsulate this logic to DBMigration ierarchy